From d8485380224c24485a4290e124d5f592c331da46 Mon Sep 17 00:00:00 2001
From: Sylvain Corlay <sylvain.corlay@gmail.com>
Date: Thu, 19 Apr 2018 22:41:44 +0200
Subject: [PATCH] Fixup post-link

---
 ipympl/__init__.py | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/ipympl/__init__.py b/ipympl/__init__.py
index aa99171..eed7c31 100644
--- a/ipympl/__init__.py
+++ b/ipympl/__init__.py
@@ -1,10 +1,8 @@
 import sys
-import matplotlib
 from ._version import version_info, __version__
 
 npm_pkg_name = 'jupyter-matplotlib'
 
-
 def _jupyter_nbextension_paths():
     return [{
         'section': 'notebook',
@@ -20,4 +18,8 @@ if 'IPython' in sys.modules:
     from IPython.core.pylabtools import backend2gui
     backend2gui['module://ipympl.backend_nbagg'] = 'ipympl'
 
-matplotlib.use('module://ipympl.backend_nbagg')
+# __init__.py is used by the nbextension installation.
+# Conda cannot have dependencies for post-link scripts.
+if 'matplotlib' in sys.modules:
+    import matplotlib
+    matplotlib.use('module://ipympl.backend_nbagg')
-- 
2.5.3

